### PR TITLE
evaluator: change the remaining string functions.

### PR DESCRIPTION
**Pull request by zxylvlp**
Change the remaining string functions.

### CODE REVIEW

**Comment by zxylvlp** [comment]
@coocood @shenli

File: evaluator/builtin_string.go [hunk 1]
```diff
@@ -180,103 +180,114 @@ func builtinUpper(args []types.Datum, _ context.Context) (d types.Datum, err err
 }
 
 // See: https://dev.mysql.com/doc/refman/5.7/en/string-comparison-functions.html
-func builtinStrcmp(args []interface{}, _ context.Context) (interface{}, error) {
-	if args[0] == nil || args[1] == nil {
-		return nil, nil
+func builtinStrcmp(args []types.Datum, _ context.Context) (d types.Datum, err error) {
```
Comment (zimulala) [inline evaluator/builtin_string.go]: Put `d.SetNull()` here, then if `err != nil` is true , we don't need to set it again?

**Comment by zimulala** [comment]
LGTM

File: evaluator/builtin_string.go [hunk 2]
```diff
@@ -180,103 +167,101 @@ func builtinUpper(args []types.Datum, _ context.Context) (d types.Datum, err err
...
+	if args[0].Kind() == types.KindNull || args[1].Kind() == types.KindNull {
+		return d, nil
```
Comment (shenli) [inline evaluator/builtin_string.go]: Default kind for datum is KindNull?

Comment (zimulala) [inline evaluator/builtin_string.go]: yep

**Comment by shenli** [comment]
LGTM

**Pull request merged**